src/main/java/htsjdk/samtools/util/TestUtil.java

```diff
@@ -73,6 +73,20 @@ public static void recursiveDelete(final File directory) {
     }
 
     /**
+     * Schedule a file or directory to be (recursively) deleted on jvm exit.
+     *
+     * This will silently delete the directory as well as it's contents.
+     * It improves upon {@link TestUtil#recursiveDelete(File)} by deleting directories and files that did not exist at call time.
+     *
+     * It cannot be undone, dir will be deleted unless program is killed.
+     *
+     * @param dir to be deleted
+     */
+    public static void deleteRecursivelyOnExit(final File dir){
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> recursiveDelete(dir)));
```

Comment:
I think the implementation of recursive delete may not be ideal here, since it can throw.  I'm not sure what happens if a shutdown hook throws, but I imagine it's not greatest thing to do.  You probably want to use a silent delete that eats any exceptions since I think it's better to leave a file undeleted than to have the test suite die inexplicably.  It's sort of a matter of taste I think.

Reply:
It's also possible that throwing during shutdown isnt' a problem as well..